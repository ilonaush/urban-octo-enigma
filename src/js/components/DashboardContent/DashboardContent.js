import React, {Component, Suspense} from 'react';
import "./DashboardContent.styl";
import List from "../List/List";
import {Route} from "react-router-dom";
import Loader from "../Loader/Loader";


const AddWorkerForm = React.lazy(() => import(/* webpackChunkName: "addForm" */"components/AddWorkerForm/AddWorkerForm"));
const FireWorkerForm = React.lazy(() => import(/* webpackChunkName: "fireForm" */"components/FireWorkerForm/FireWorkerForm"));
const Gallery = React.lazy(() => import(/* webpackChunkName: "gallery" */"components/Gallery/Gallery.js"));


export class DashboardContent extends Component {


    render() {
        return (
            <div className='d-content'>
                <Route exact path='/' component={List}/>
                <Route  exact path='/add-worker' render={(routeProps) =>  <Suspense fallback={<Loader/>}>
                    <AddWorkerForm handleLoading = {this.props.handleLoading} {...routeProps}/></Suspense>}/>
                <Route  exact path='/fire-worker' render={(routeProps) => <Suspense fallback={<Loader/>}>
                    <FireWorkerForm handleLoading = {this.props.handleLoading} {...routeProps}/></Suspense>}/>
                <Route  exact path='/gallery' render={(routeProps) => <Suspense fallback={<Loader/>}>
                    <Gallery handleLoading = {this.props.handleLoading} {...routeProps}/></Suspense>}/>
            </div>
        );
    }
}

DashboardContent.propTypes = {};

export default DashboardContent;
