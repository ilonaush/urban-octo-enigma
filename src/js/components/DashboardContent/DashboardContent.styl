.d-content
  background-color white
  min-height 600px
  width 98%
  margin -28px auto 0 auto
  border-radius .1875rem
  box-shadow 0 1px 15px 1px rgba(39,39,39,.1)
