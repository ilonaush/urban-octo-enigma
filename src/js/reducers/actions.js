import RequestService from "../services/RequestService";
import {ACTIONS_TYPES, LOADING_TYPES, REQUEST_PATHS} from "../constants";

/**
 * sets loading state and loader type
 * @param type
 * @param dispatch
 * @returns {*}
 */
function setLoading(type, dispatch) {
    return dispatch({
        type: ACTIONS_TYPES.SET_LOADING,
        payload: {loading: true, loadingType: type}
    });
}

/**
 * dispatches empty array
 * @param dispatch
 * @param type
 */
function handleError(dispatch, type) {
    dispatch({
        type: type,
        payload: { cats: [], loading: false }
    })
}

export default {
    getCats: function() {
        return  async (dispatch) => {

            setLoading(null, dispatch);

            try {
                const {data: cats} = await RequestService.get(REQUEST_PATHS.GET_CATS);
                dispatch({
                    type: ACTIONS_TYPES.GET_CATS,
                    payload: { cats, loading: false }
                })
            }
            catch (e) {
              handleError(dispatch, ACTIONS_TYPES.GET_CATS)
            }
        };
    },
    addCat: function (cat) {
        return async (dispatch) => {

            setLoading(null, dispatch);

            try {
                const {data: {cats, status}} = await RequestService.post(REQUEST_PATHS.ADD_CAT, cat);
                if (!status) {
                    handleError(dispatch, ACTIONS_TYPES.ADD_CAT)
                }
                dispatch({
                    type: ACTIONS_TYPES.ADD_CAT,
                    payload: {cats, loading: false}
                });
            }
            catch (e) {
                handleError(dispatch, ACTIONS_TYPES.ADD_CAT)
            }
        }
    },
    feedCat(catInfo) {
        return async (dispatch) =>  {

            setLoading(LOADING_TYPES.FEED_CAT, dispatch);

            try {
                const {data: {cats, status}} = await RequestService.patch(REQUEST_PATHS.FEED_CAT, catInfo);
                if (!status) {
                    handleError(dispatch, ACTIONS_TYPES.FEED_CAT)
                }
                dispatch({
                    type: ACTIONS_TYPES.FEED_CAT,
                    payload: {cats, loading: false}
                });
            }
            catch (e) {
                handleError(dispatch, ACTIONS_TYPES.FEED_CAT)

            }
    }},
    hugCat(catInfo) {
        return async (dispatch) =>  {

            setLoading(LOADING_TYPES.HUG_CAT, dispatch);

            try {
                const {data: {cats, status}} = await RequestService.patch(REQUEST_PATHS.HUG_CAT, catInfo);
                if (!status) {
                    handleError(dispatch, ACTIONS_TYPES.HUG_CAT)
                }
                dispatch({
                    type: ACTIONS_TYPES.HUG_CAT,
                    payload: {cats, loading: false}
                });
            }
            catch (e) {
                handleError(dispatch, ACTIONS_TYPES.HUG_CAT)
            }
    }},
    washCat(catInfo) {
        return async (dispatch) =>  {

            setLoading(LOADING_TYPES.WASH_CAT, dispatch);

            try {
                const {data: {cats, status}} = await RequestService.patch(REQUEST_PATHS.WASH_CAT, catInfo);
                if (!status) {
                    handleError(dispatch, ACTIONS_TYPES.WASH_CAT)
                }
                dispatch({
                    type: ACTIONS_TYPES.WASH_CAT,
                    payload: {cats, loading: false}
                });
            }
            catch (e) {
                handleError(dispatch, ACTIONS_TYPES.WASH_CAT)
            }
    }},
    findHome: function (cat) {
        return async (dispatch) =>  {

            setLoading(null, dispatch);

            try {
                const variables = {
                    ...cat
                };

                const {data: {cats, status}} = await RequestService.patch(REQUEST_PATHS.ISSUE_CAT, variables);
                if (!status) {
                    handleError(dispatch, ACTIONS_TYPES.ISSUE_CAT)
                }
                dispatch({
                    type: ACTIONS_TYPES.ISSUE_CAT,
                    payload: {cats, loading: false}
                });
            }
            catch (e) {
                handleError(dispatch, ACTIONS_TYPES.ISSUE_CAT)
            }
        }
    }
};
