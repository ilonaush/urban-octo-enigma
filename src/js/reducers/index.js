import { createStore, applyMiddleware } from 'redux';
import reducer   from './reducers';
import  { saveStore, loadStore } from '../services/LocalStorageService';
import thunk from 'redux-thunk'
import { createLogger } from 'redux-logger'

const loggerMiddleware = createLogger();

// const predefinedStore = loadStore();

const store = createStore(
    reducer,
    // predefinedStore,
    applyMiddleware(thunk, loggerMiddleware)
);

// store.subscribe(() => {
//     saveStore(store.getState())
// });

export {
    store,
    // predefinedStore
};
