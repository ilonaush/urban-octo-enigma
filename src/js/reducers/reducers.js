import {ACTIONS_TYPES} from "../constants";

const initialState = {
  loading: false,
};

let reducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS_TYPES.GET_CATS:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ADD_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.FEED_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.HUG_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.WASH_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ISSUE_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.SET_LOADING: {
      return {
        ...state,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      }
    }
    default:
      return state;
  }
};

export default reducer;
