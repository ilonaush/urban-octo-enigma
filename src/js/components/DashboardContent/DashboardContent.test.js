import React from 'react';

import DashboardContent  from './DashboardContent';

import ReactDOM from 'react-dom';
import {MemoryRouter} from "react-router";
import {Provider} from "react-redux";
import {store} from "../../reducers/index";


describe('DashboardContent', () => {
    it('renders', () => {

        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter><Provider store={store}><DashboardContent/></Provider></MemoryRouter>, div)
    })
})