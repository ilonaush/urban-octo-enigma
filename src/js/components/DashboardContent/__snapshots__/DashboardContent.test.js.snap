// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardContent renders correctly 1`] = `
<div
  className="d-content"
>
  <div
    className="loader"
  >
    <img
      alt=""
      src="/images/loaders/blue-loader.gif"
    />
  </div>
</div>
`;
